File: back/src/models/ServiceYear.js
import mongoose from 'mongoose';

const serviceYearSchema = new mongoose.Schema(
  {
    year: { type: String, required: true, unique: true, index: true },
    isActive: { type: Boolean, default: false },
    archivedAt: { type: String, default: null },
  },
  {
    timestamps: true,
    strict: false,
    collection: 'serviceYears',
  }
);

const ServiceYear = mongoose.models.ServiceYear || mongoose.model('ServiceYear', serviceYearSchema);
export default ServiceYear;

File: back/Models/ServiceYear.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Eftekad.Backend.Models
{
    [Table("serviceYears")]
    public class ServiceYear
    {
        [Key]
        [Column("year")]
        public string Year { get; set; } = string.Empty;

        [Column("isActive")]
        public bool IsActive { get; set; } = false;

        [Column("archivedAt")]
        public string? ArchivedAt { get; set; }

        [Column("createdAt")]
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        [Column("updatedAt")]
        public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
    }
}
